src/Workspaces/Core/Portable/Workspace/Solution/Checksum.cs

```diff
@@ -136,6 +136,12 @@ public override unsafe string ToString()
         public static bool operator !=(Checksum left, Checksum right)
             => !(left == right);
 
+        public static bool operator ==(Checksum left, HashData right)
+            => left._checksum == right;
+
+        public static bool operator !=(Checksum left, HashData right)
+            => !(left == right);
```

Comment:
😕 how/why is `HashData` not private?

Reply:
I hoenstly don't understand the design between checksum/hashdata.  the former seems to exist just to add a heap allocation on every checksum we have.

Reply:
no clue.  it's a bit of a wonky design imo.

Reply:
I would prefer an approach that doesn't further expose `HashData` (ideally we make the type `private`)

Reply:
Huh... just realized `Checksum` is a reference type. How unfortunate.